Client/Frontend/Browser/URLBarView.swift

```diff
@@ -406,6 +407,17 @@ class URLBarView: UIView {
         }
         updateViewsForOverlayModeAndToolbarChanges()
     }
+    
+    func setShowTopTabs(topTabsIsShowing: Bool) {
```

Comment:
Instead of having a separate setter method, you can override add a `didSet` block to `topTabsIsShowing`:

```swift
var topTabsIsShowing = false {
    didSet {
        curveShape.hidden - topTabsIsShowing
        updateViewsForOverlayModeAndToolbarChanges()
    }
}
```

I'm also not sure if we need to call `updateViewsForOverlayModeAndToolbarChanges` since it just configures the hidden booleans for the toolbar buttons and nothing to do with the curve.